### DISCUSSION TITLE
region-in event still happens, after clearRegions

### DISCUSSION THREAD

**Discussion by ApayRus** [katspaugh/wavesurfer.js#3469] [Q&A]
Hi. 
v6. 
I remove regions with clearRegions() method. Visually regions disappears from waveform.
But 'region-in' event still happens , like they wasn't deleted. 
Any ideas please? 

UPD:
I did ws.un('region-in'), ws.regions.un('region-in') . 
Anyway events happens, as all old phrases are on the waveform...

**Comment by katspaugh** [comment]
Hmm, it's supposed to stop emitting that event on remove: https://github.com/katspaugh/wavesurfer.js/blob/v6/src/plugin/regions/region.js#L382-L389

Try adding a breakpoint one of the lines I linked and see if an unsubscription happens on remove.

**Reply by ApayRus** [reply]
They both fires. Line 389 on remove region. 
And then line 382 on 'region-in' (also after removing). 
What I have noticed, 'region-in' doesn't fire on 'seek', only on 'playing'. 
![Screenshot from 2024-01-08 04-42-14](https://github.com/katspaugh/wavesurfer.js/assets/1222611/3d78fb83-b2ed-43be-bb3c-2f28f66c35dd)

**Reply by ApayRus** [reply]
i imported runinng project to codesandbox 
[Demo](https://zlslmy-1234.csb.app/), in a console you can see firing 'region-in' after removing region. 
[Code](https://codesandbox.io/p/github/ApayRus/react-wavesurfer-provider/master?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clr49nux600062v6kcxeuouc4%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clr49nux600022v6kcxcko0c5%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clr49nux600042v6kw9t960eo%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clr49nux600052v6kimv4j217%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B51.48293438468765%252C48.51706561531235%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clr49nux600022v6kcxcko0c5%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr49nux500012v6k49zj0rjv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clr49nux600022v6kcxcko0c5%2522%252C%2522activeTabId%2522%253A%2522clr49nux500012v6k49zj0rjv%2522%257D%252C%2522clr49nux600052v6kimv4j217%2522%253A%257B%2522id%2522%253A%2522clr49nux600052v6kimv4j217%2522%252C%2522activeTabId%2522%253A%2522clr4af84b01302v6k5sg5dn6y%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A1234%252C%2522id%2522%253A%2522clr4af84b01302v6k5sg5dn6y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%257D%252C%2522clr49nux600042v6kw9t960eo%2522%253A%257B%2522id%2522%253A%2522clr49nux600042v6kw9t960eo%2522%252C%2522activeTabId%2522%253A%2522clr4adj47010g2v6k9s3adivl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clr49nux600032v6krcoepr4o%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clr49nwwr0012egk5dg989ew5%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522clr49nybb00572v6k1t8m69pa%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Ftypescript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522clr49thud008q2v6kzlglcqm1%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522clr4adj47010g2v6k9s3adivl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clr4adj71001qegk5bsqd5ie7%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A22.24390216943128%257D)

**Reply by katspaugh** [reply]
It's a lot of code to dig through. It would be nice to create a minimal React-free example demonstrating the problem.

The only thing I noticed was that you're [unsubscribing from `region-in` in the wrong way](https://codesandbox.io/p/github/ApayRus/react-wavesurfer-provider/master?file=%2Fsrc%2FProvider.tsx%3A168%2C1-169%2C1):

```js
        wavesurferRef.current.regions.un('region-in', () => {});
```

Instead of `() => {}`, the original event handler that was passed to `on()` should be passed.

As a side note, I'm curious about what is stopping you from upgrading to v7?

**Reply by ApayRus** [reply]
Thank you for answer. 
I hope you have seen, that regions removed, but events still happen... 
I plan to upgrade to v7. 
`contenteditable` was the reason to stay with v6  ). 
I saw it appeared in v7 a couple of days before :)

**Reply by ApayRus** [reply]
> As a side note, I'm curious about what is stopping you from upgrading to v7?

`region.play()` doesn't stop playing the audio when `region.end` reached. 

As a workaround, I did: 
```
regions.on('region-out', () => {
					waveSurferRef.current?.pause()
				})
```
But then, if the region finished, you cannot play another by click.
